src/api-v1.js

```diff
@@ -63,35 +64,35 @@ api.declare({
   let logthingy = `${url} in ${service}/${region}`;
   debug(`Attempting to redirect to ${logthingy}`);
 
+  try {
+    await validateInputUrl(url, this.allowedPatterns, this.redirectLimit, this.ensureSSL);
+  } catch (err) {
+    debug(err.stack || err);
+    // Intentionally vague because we don't want to reveal
+    // our configuration too much
+    let msg = 'Input URL failed validation for unknown reason';
```

Comment:
I've renamed the function to followRedirects, and in that case all of these exceptions are valid exceptions.  We just coincidentally use the followRedirects function to also check for validity of input urls.  Fwiw, followRedirects was the name of this function as I had originally written it, not sure why I changed it to validateInputUrl.

The primary purpose of this function is to actually follow the redirect chain in a special way and pass the final URL on to the function that creates the read stream for uploading.